test/base/Algorithms.test.cc

```diff
@@ -17,9 +17,7 @@
 TEST(AlgorithmsTest, minmax)
 {
     EXPECT_EQ(1, celeritas::min<int>(1, 2));
-    EXPECT_NE(0.2, celeritas::min<float>(0.1, 0.2));
```

Comment:
Out of curiosity, what was the warning message? (And is the EXPECT_EQ/NE float safe or should those be EXPECT_SOFT_NEAR?)

Reply:
There are two warnings here -- the first is that `celeritas::min<float>(0.1, 0.2)` is implicitly converting double-precision rvalue constants (`0.1` is double) into floats. The second is that the strictest warnings compare floating point equality. Here exact equality would be safe if it compared `0.2f`  on the left, but for a general double value `v`,
```c++
(v  == double(float(v)))
```
is not always true. And in fact, this test would still "pass" even if you did `min<float>(0.3, 0.2)`, because
```c++
double d = 0.20000000000000001;
float f = 0.200000003;
double df = 0.20000000298023224;
```
so the test here is totally bogus. The second `int` test actually makes sure the implementation isn't always returning the first of the two elements anyway ;)

Reply:
Fair enough.  The spirit of the test would have been kepts with:
```
EXPECT_LT(0.15, celeritas::min<float>(0.1f, 0.2f));
```
or
```
EXPECT_GT(0.15, celeritas::min(0.1, 0.2));
```
(if this compile it would the double version).

Reply:
True, but again it would still  be redundant with the `int` test which doesn't have such ambiguities. Besides, your test there reminds me of  a certain `if (z < 1.5)` that we saw once long ago 😂